platform/linux-generic/arch/x86/odp_time_cpu.c

```diff
@@ -15,24 +14,67 @@
 
 #include <time.h>
 
-/* Measure TSC frequency. Frequency information registers are defined for x86,
- * but those are often not enumerated. */
+static int nsleep(uint64_t nsec)
+{
+	struct timespec ts1, ts2;
+	uint64_t diff;
+
+	if (clock_gettime(CLOCK_MONOTONIC_RAW, &ts1)) {
+		_ODP_ERR("clock_gettime() failed\n");
+		return 1;
+	}
+
+	do {
+		if (clock_gettime(CLOCK_MONOTONIC_RAW, &ts2)) {
+			_ODP_ERR("clock_gettime() failed\n");
+			return 1;
+		}
+
+		diff = (ts2.tv_sec - ts1.tv_sec) * ODP_TIME_SEC_IN_NS +
+			ts2.tv_nsec - ts1.tv_nsec;
+	} while (diff < nsec);
+
+	return 0;
+}
+
+static void sort(uint64_t values[], int num)
+{
+	for (int n = 0; n < num; n++) {
+		for (int i = n + 1; i < num; i++) {
+			if (values[i] < values[n]) {
+				uint64_t tmp = values[i];
+
+				values[i] = values[n];
+				values[n] = tmp;
+			}
+		}
+	}
+}
+
+static uint64_t median(uint64_t values[], int num)
+{
+	sort(values, num);
```

Comment:
Use `qsort()`?

Reply:
I thought about initially it but ended up not using it as performance is not a factor and I thinkt the simple sort is as (or more?) readable in this case.  Did not change this in v2.

Reply:
Slightly less code to maintain with `qsort()`, anyways, resolving this.